src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key] as string;
+
+    if (!result[keyValue]) {
+      result[keyValue] = [];
+    }
+
+    result[keyValue].push(item);
+
+    return result;
```

Comment:
Using an assertion to cast the initial value of the reduce function to `GroupsMap<T>` can be error-prone if the items array is empty or if the key values are not strings. It would be safer to initialize the result with an explicit type or to provide a more robust way to initialize the accumulator to match the `GroupsMap<T>` type.